util.c

```diff
@@ -1658,6 +1662,9 @@ x_readlink(const char *path)
 }
 #endif
 
+/* Avoid breaking the api */
```

Comment:
Yeah, that would be a more reasonable solution. Perhaps refactor read_file/read_text_file ?